core/Rename.cpp

```diff
@@ -132,10 +132,9 @@ namespace olympia
         }
 
         const auto & srcs = inst_ptr->getRenameData().getSourceList();
-        if(inst_ptr->isStoreInst() && srcs.size() < 2 && dests.size() == 0){
-            // free PRF for STORE instruction used for data tracking
-            core_types::RegFile rf = core_types::RF_INTEGER;
-            freelist_[rf].push(inst_ptr->getRenameData().getAddrReg().val);
+        if(inst_ptr->isStoreInst() && dests.size() == 0){
+            // decrement counter for store data tracking
+            --reference_counter_[inst_ptr->getRenameData().getDataReg().rf][inst_ptr->getRenameData().getDataReg().val];
```

Comment:
Wow, that's a big line!  😆 Can you break it up?
```suggestion
            const auto & data_reg = inst_ptr->getRenameData().getDataReg();
            --reference_counter_[data_reg.rf][data_reg.val];
```